src/util/BaseLoginController.js

```diff
@@ -101,6 +101,12 @@ define(['okta', 'q'], function (Okta, Q) {
     postRenderAnimation: function() {
       // Event triggered after a page is rendered along with the classname to identify the page
       this.trigger('pageRendered', {page: this.className});
+      var context = {
+        view: window.location.pathname,
+        controller: this.className,
+        transaction: this.options.appState.get('transaction')
+      };
+      this.trigger('navigated', _.pick(context, _.identity));
```

Comment:
This will remove the `transaction` key if it is `undefined`. Figured this was a cleaner solution to:

```javascript
var context = {
  view: window.location.pathname,
  controller: this.className
};

var transaction = this.options.appState.get('transaction');
if (transaction) {
  context.transaction = transaction;
}

this.trigger('navigated', context);
```

Reply:
hmm, why would remove `transaction` when it's `undefined`?
It seems doesn't hurt and by keeping it's a little easier to express the type of `context`

Reply:
Mostly a style thing - I don't have a set opinion on this, but do like that I will not need to account for this undefined values in my tests :)

Per your suggestion, should I simplify this method to:

 ```javascript
var context = {
  view: window.location.pathname,
  controller: this.className,
  transaction: this.options.appState.get('transaction')
};

this.trigger('navigated', context);

// Tests
expect(test.navigatedSpy).toHaveBeenCalledWith({
  view: '/context.html',
  controller: 'primary-auth',
  transaction: undefined
});
```

Reply:
gotcha. either way works for me and slightly prefer to keep `undefined`.